private/pkg/protosource/protosource.go

```diff
@@ -152,10 +152,9 @@ type OptionExtensionDescriptor interface {
 	// have a set value on this descriptor.
 	PresentExtensionNumbers() []int32
 
-	// PresentFields returns a map of field descriptor to value for all
-	// option fields present. This includes both normal option fields
-	// and extension fields (aka custom options).
-	PresentFields() map[protoreflect.FieldDescriptor]protoreflect.Value
+	// Range implements the same interface as protoreflect.Message. This takes
+	// in a callback that iterates over a protoreflect message field descriptors and values.
```

Comment:
I'd focus more on what the method does vs. comparing to protoreflect.Message.
```suggestion
	// Range iterates through all fields present in the options message invoking
	// the callback for each one. If the callback returns false, it terminates the
	// iteration and returns immediately.
```

Reply:
+1, see comment below